### ISSUE TITLE
Move transpile tests into code

### ISSUE THREAD

**Issue by syxolk** [mp4096/indentex#11]
What do you think about moving the test cases from `tests/test_cases/*` to `src/transpile.rs`?

Main advantages:
- Better test coverage :)
- We can also test the recently added `--flatten-output`

**Comment by mp4096** [comment]
Sure, they absolutely belong to the source in the Rust convention of unit tests being in the same file as the code.

Current tests in `./tests/test_cases/` are stuck between integration and unit test. I'd rather have fine-grained, per-feature unit tests in `transpile.rs` and a real-life corpus as integration test.

However, I'd rather write unit tests for all lower-level parsers first. The thing is, as soon as you add integration tests, coverage mostly goes to 90-95% and you cannot see anymore whether your lower-level stuff is covered as well because everything is green.

Coverage is only as good as unit tests are. My [Blockschaltbilder](https://github.com/mp4096/blockschaltbilder) boilerplate generator has ≈ 94% coverage (uncovered 6% are file IO). And I had a [bug](https://github.com/mp4096/blockschaltbilder/commit/380dcffe0b70ebec5b3eaffd206f24115e77a270) lurking in there. This line was covered by an integration test, but not by a unit test.

---

TL;DR Yes, but only after we have unit tests for all helper functions.

**Comment by mp4096** [comment]
In a way, solved. One the one hand I've removed the overall "end-to-end" test using Python, on the other hand the lib has 100% coverange now.
